packages/ui/src/proposals/modals/AddNewProposal/components/SpecificParameters/DecreaseWorkingGroupLeadStake.tsx

```diff
@@ -0,0 +1,118 @@
+import BN from 'bn.js'
+import React, { useEffect } from 'react'
+
+import { InputComponent, InputNumber } from '@/common/components/forms'
+import { Info } from '@/common/components/Info'
+import { AmountButton, AmountButtons, Row, TransactionAmount } from '@/common/components/Modal'
+import { RowGapBlock } from '@/common/components/page/PageContent'
+import { TextInlineMedium, TextMedium } from '@/common/components/typography'
+import { capitalizeFirstLetter } from '@/common/helpers'
+import { useNumberInput } from '@/common/hooks/useNumberInput'
+import { formatTokenValue } from '@/common/model/formatters'
+import { SelectMember } from '@/memberships/components/SelectMember'
+import { useMember } from '@/memberships/hooks/useMembership'
+import { SelectWorkingGroup } from '@/working-groups/components/SelectWorkingGroup'
+import { useWorkingGroup } from '@/working-groups/hooks/useWorkingGroup'
+
+export interface DecreaseWorkingGroupLeadStakeParameters {
+  stakingAmount?: BN
+  groupId?: string
+  workerId?: number
+}
+
+interface DecreaseWorkingGroupLeadStakeProps extends DecreaseWorkingGroupLeadStakeParameters {
+  setStakingAmount: (amount: BN) => void
+
+  setGroupId(groupId: string): void
+
+  setWorkerId(workerId?: number): void
+}
+
+export const DecreaseWorkingGroupLeadStake = ({
+  stakingAmount,
+  groupId,
+  setStakingAmount,
+  setGroupId,
+  setWorkerId,
+}: DecreaseWorkingGroupLeadStakeProps) => {
+  const [amount, setAmount] = useNumberInput(0, stakingAmount)
+  const setHalf = () => setAmount(stakingAmount ? stakingAmount.divn(2).toString() : '')
+  const setThird = () => setAmount(stakingAmount ? stakingAmount.divn(3).toString() : '')
+
```

Comment:
This should use current working group leader stake. So if current WG Lead stake is 6000 JOY:
- set half will always set 3000 JOY
- set third will always set 2000 JOY
The logic (callback) will change when the group is changed.

Reply:
✅